language/move-vm/types/src/values/values_impl.rs

```diff
@@ -1456,9 +1456,9 @@ pub mod vector {
             (Type::U128, Container::U128(v)) => v.push(e.value_as()?),
             (Type::Bool, Container::Bool(v)) => v.push(e.value_as()?),
 
-            (Type::Struct(_), Container::General(v)) | (Type::Address, Container::General(v)) => {
-                v.push(e.0)
-            }
+            (Type::Struct(_), Container::General(v))
```

Comment:
The `match (&ty_args[0], &mut *v) {` pattern is a bit hard to read + is duplicated in each vector native function. The problem is that it's doing two distinct things:
(1) Type-checking the type argument against the container type
(2) Extracting a `Vec<T>` from the container so the native function can do something with it (e.g., call `push`)

 I wonder if we could improve this code + reduce duplication by adding utility functions for these?

Reply:
Let me experiment with it. I'll put up a separate diff for this.